### PR TITLE
[RLlib] Change default framework from tf to torch

### PR DESCRIPTION
**Pull request by kouroshHakha**
## Why are these changes needed?
This PR changes the default framework_str from tf to either torch or tf2. First step towards hopefully deprecating tf1 stack. 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/algorithms/algorithm_config.py [hunk 1]
```diff
@@ -261,7 +261,7 @@ def __init__(self, algo_class=None):
         self.placement_strategy = "PACK"
 
         # `self.framework()`
-        self.framework_str = "tf"
+        self.framework_str = "torch"
```
Comment (sven1977) [inline rllib/algorithms/algorithm_config.py:264]: :fingers-crossed: :)

**Review by sven1977 — APPROVED** [review]
Looks great! Some example scripts used to run only on tf and will now only run on torch, but I guess that's ok. E.g. `custom_metrics_and_callbacks`.

File: rllib/examples/documentation/saving_and_loading_algos_and_policies.py [hunk 2]
```diff
@@ -297,6 +298,6 @@ def new_policy_mapping_fn(agent_id, episode, worker, **kwargs):
 
 # __export-models-as-onnx-begin__
 # Using the same Policy object, we can also export our NN Model in the ONNX format:
-ppo_policy.export_model("/tmp/my_nn_model", onnx=True)
+ppo_policy.export_model("/tmp/my_nn_model", onnx=False)
```
Comment (gjoliver) [inline rllib/examples/documentation/saving_and_loading_algos_and_policies.py:301]: update comment?

File: rllib/BUILD [hunk 3]
```diff
@@ -1618,7 +1618,7 @@ py_test(
 py_test(
     name = "connectors/tests/test_agent",
     tags = ["team:rllib", "connector"],
-    size = "small",
+    size = "medium",
```
Comment (gjoliver) [inline rllib/BUILD:1621]: this is from some other pr right?

File: rllib/env/tests/test_multi_agent_env.py [hunk 4]
```diff
@@ -449,6 +449,7 @@ def compute_actions_from_input_dict(
             env_creator=lambda _: MultiAgentCartPole({"num_agents": 2}),
             default_policy_class=ModelBasedPolicy,
             config=DQNConfig()
+            .framework("tf")
```
Comment (gjoliver) [inline rllib/env/tests/test_multi_agent_env.py:452]: curious, multi-agent env doesn't work with torch?

File: rllib/BUILD (hunk 3 above)
Comment (kouroshHakha) [reply rllib/BUILD:1621]: The other PR will get merged and this difference will go away. I wanted to make sure the tests on CI doesn't get red b/c of time outs.

**Pull request merged**